### PR TITLE
populate RFI information from burst's `burst_rfi_info`

### PR DESCRIPTION
**Pull request by seongsujeong**
This PR is the follow-up of #115 and #101 to populate the RFI information based on `Sentinel1BurstSlc.burst_rfi_info`.
This PR was tested on three cases, 
- S1 SAFE with RFI detected & mitigated:
  - `S1A_IW_SLC__1SDV_20230108T135249_20230108T135316_046693_0598D3_BA76.zip`
  - In this case, `rfi_burst_report`, `rfi_burst_report/frequency_domain_rfi_report` and/or `rfi_burst_report/time_domain_rfi_report` gets populated

- S1 SAFE with RFI information but no mitigation was detected nor performed:
  - `S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.zip`
  - `rfi_burst_report` gets populated, but time & frequency rfi report does not get populated.

- S1 SAFE with old IPF version (`2.36` in case of the example below):
  - `S1A_IW_SLC__1SSV_20150207T014953_20150207T015023_004511_005899_9C44.zip`
  - Its IPF version is `2.36`. Note that the RFI annotation is available from IPF >= `3.40`
  - RFI annotation is not available in the old data. In this case, `is_rfi_info_available` is `False` and none of the information gets populated.

### CODE REVIEW

File: src/compass/s1_cslc_qa.py [hunk 1]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+            # create HDF5 group for RFI info
+            h5_rfi_path = f'{QA_PATH}/rfi_information/{burst.polarization}'
+            rfi_group = cslc_h5py_root.require_group(h5_rfi_path)
+
+            # write items to HDF5 and dict
+            _qa_items_to_h5_and_dict(rfi_group, self.rfi_dict, rfi_qa_items_pol)
+
+            # Take care of the burst RFI report information
+            if not is_rfi_info_available:
+                return
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:321]: `rfi_mitigation_performed` and `rfi_mitigation_domain` are available if `is_rfi_info_available` is `False`?

File: src/compass/s1_cslc_qa.py [hunk 2]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+                Meta('swath',
+                     rfi_burst_report['swath'],
+                     'Swath of the burst'),
+                     Meta('azimuth_time',
+                          datetime.datetime.strftime(rfi_burst_report['azimuthTime'],
+                                                     TIME_STR_FMT),
+                          'Azimuth time of the burst report'),
+                     Meta('in_band_out_band_power_ratio',
+                          rfi_burst_report['inBandOutBandPowerRatio'],
+                          'Ratio between the in-band and out-of-band power of the burst')
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:337]: Alignment is off in this list.

File: src/compass/s1_cslc_qa.py [hunk 3]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+                rfi_info_list = [Meta('rfi_mitigation_performed',
+                                      burst.burst_rfi_info.rfi_mitigation_performed,
+                                      ('Activation strategy of RFI mitigation'
+                                      '["never", "BasedOnNoiseMeas","always"]')),
+                                 Meta('rfi_mitigation_domain',
+                                      burst.burst_rfi_info.rfi_mitigation_domain,
+                                      'Domain the RFI mitigation step was performed')]
+                rfi_qa_items_pol += rfi_info_list
+
+            # create HDF5 group for RFI info
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:312]:
```suggestion
            # create HDF5 group for RFI info for current polarization
```
Clarity

File: src/compass/s1_cslc_qa.py [hunk 4]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+                    Meta('percentage_affected_lines',
+                         time_domain_report['percentageAffectedLines'],
+                         'Percentage of level-0 lines affected by RFI.'),
+                    Meta('avg_percentage_affected_samples',
+                         time_domain_report['avgPercentageAffectedSamples'],
+                         'Average percentage of affected level-0 samples in the lines containing RFI'),
+                    Meta('max_percentage_affected_samples',
+                         time_domain_report['maxPercentageAffectedSamples'],
+                         'Maximum percentage of level-0 samples affected by RFI in the same line'),
+                ]
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:359]:
```suggestion
                ]

```
nit - Add a line for consistency with all the other metadata add blocks.

File: src/compass/s1_cslc_qa.py [hunk 5]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+            rfi_qa_items_pol = [Meta('is_rfi_info_available',
+                                is_rfi_info_available,
+                                'Whether or not RFI information is available')]
+
+            if is_rfi_info_available:
+                # Follow key/values only assigned if RFI info is avaiable
+                rfi_info_list = [Meta('rfi_mitigation_performed',
+                                      burst.burst_rfi_info.rfi_mitigation_performed,
+                                      ('Activation strategy of RFI mitigation'
+                                      '["never", "BasedOnNoiseMeas","always"]')),
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:306]:
```suggestion
                                      '["never", "BasedOnNoiseMeas", "always"]')),
```

File: src/compass/s1_cslc_qa.py [hunk 6]
```diff
@@ -279,32 +280,134 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+
+                self.rfi_dict['rfi_burst_report']['frequency_domain_rfi_report'] = {}
+                rfi_burst_report_freq_domain_group = rfi_burst_report_group.require_group('frequency_domain_rfi_report')
+                _qa_items_to_h5_and_dict(rfi_burst_report_freq_domain_group,
+                                        self.rfi_dict['rfi_burst_report']['frequency_domain_rfi_report'],
+                                        burst_freq_domain_report_item)
+
+                # Take care of isolated RFI report inside frequency burst RFI report
+                isolated_rfi_report = freq_domain_report['isolatedRfiReport']
+                # ['percentageAffectedLines', 'maxPercentageAffectedBW']
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:396]:
```suggestion
```
comment clean up

**Review by LiangJYu — APPROVED** [review]
LGTM. My comments are cosmetic.

**Review by vbrancat — APPROVED** [review]
LGTM

File: src/compass/s1_cslc_qa.py (hunk 1 above)
Comment (seongsujeong) [reply src/compass/s1_cslc_qa.py:321]: No. `rfi_mitigation_performed` and `rfi_mitigation_domain` are available only when RFI annotation exists.
If the RFI annotation do not exist, then `is_rfi_info_available` is `False`

File: src/compass/s1_geocode_metadata.py [hunk 7]
```diff
@@ -149,7 +149,7 @@ def run(cfg, burst, fetch_from_scratch=False):
             cslc_qa.compute_static_layer_stats(h5_obj, cfg.rdr2geo_params)
             cslc_qa.shadow_pixel_classification(h5_obj)
             cslc_qa.set_orbit_type(cfg, h5_obj)
-            if cslc_qa.output_to_json:
+            if cfg.quality_assurance_params.output_to_json:
```
Comment (seongsujeong) [inline src/compass/s1_geocode_metadata.py:152]: @LiangJYu Can you check if this change makes sense? Thanks

File: src/compass/s1_geocode_slc.py [hunk 8]
```diff
@@ -239,10 +239,10 @@ def run(cfg: GeoRunConfig):
                         geo_burst_h5, apply_tropo_corrections,
                         cfg.tropo_params.delay_type)
                 cslc_qa.compute_CSLC_raster_stats(geo_burst_h5, bursts)
-                cslc_qa.populate_rfi_dict(geo_burst_h5)
+                cslc_qa.populate_rfi_dict(geo_burst_h5, bursts)
                 cslc_qa.valid_pixel_percentages(geo_burst_h5)
                 cslc_qa.set_orbit_type(cfg, geo_burst_h5)
-                if cslc_qa.output_to_json:
+                if cfg.quality_assurance_params.output_to_json:
```
Comment (seongsujeong) [inline src/compass/s1_geocode_slc.py:245]: @LiangJYu Can you check if this change makes sense? Thanks

File: src/compass/s1_cslc_qa.py [hunk 9]
```diff
@@ -279,32 +280,136 @@ def valid_pixel_percentages(self, cslc_h5py_root):
...
+            rfi_burst_report_list = [
+                Meta('swath',
+                     rfi_burst_report['swath'],
+                     'Swath of the burst'),
+                Meta('azimuth_time',
+                     datetime.datetime.strftime(rfi_burst_report['azimuthTime'],
+                                                TIME_STR_FMT),
+                    'Azimuth time of the burst report'),
+                Meta('in_band_out_band_power_ratio',
+                          rfi_burst_report['inBandOutBandPowerRatio'],
```
Comment (vbrancat) [inline src/compass/s1_cslc_qa.py:338]: BTW, for CSLC we decided to go the snake case (e.g. `in_band_out_band_power_ratio`) and not to use the camel case. Can you please modify this accordingly? Thanks

Comment (seongsujeong) [reply src/compass/s1_cslc_qa.py:338]: The `rfi_burst_report` is from `Sentinel1BurstSlc.burst_rfi_info`. In s1reader side, I decided to use camel case (i.e. what S1 annotations follow) to simplify the XML parsing routine.

**Pull request merged**